tools/docs.py

```diff
@@ -439,8 +439,9 @@ def _test(obj):
             except subprocess.CalledProcessError:
                 print("Failed to run documentation example:\n\n%s\n\n" % ex.source,
                       file=sys.stderr)
+                sys.stdout.flush()
```

Comment:
Should this be `sys.stderr`?

Reply:
Actually stdout on purpose here, to flush the preceding "print" statement so error report outputs show up sequenced with other ongoing output instead of (confusingly) some time later.

Reply:
The preceding print is to stderr though. If the issue is confusing interleaving of stdout and stderr, shouldn’t the stdout flush happen before the print? (And then you might want an additional stderr flush after, depending on whether Python line buffers or block buffers sys.stderr.) Alternatively, I think you could avoid confusing interleavings by running the interpreter with the `-u` flag.